exercises/pov/src/Example.hs

```diff
@@ -1,40 +1,40 @@
 module POV (Graph(Graph), fromPOV, tracePathBetween) where
 
-import Data.Maybe(listToMaybe, catMaybes)
+import Data.Maybe(listToMaybe, mapMaybe)
 
 data Graph a = Graph { value :: a, children :: [Graph a] } deriving (Show, Eq)
 data Crumb a = Crumb a [Graph a] [Graph a] deriving (Show, Eq)
-data Zipper a = Zipper { node :: (Graph a), path ::  [Crumb a] }
+data Zipper a = Zipper { node :: Graph a, path :: [Crumb a] }
 
 fromPOV :: Eq a => a -> Graph a -> Maybe (Graph a)
-fromPOV x = (fmap reparent) . (findLoc x) . rootZipper
+fromPOV x = fmap reparent . findLoc x . rootZipper
 
 tracePathBetween :: Eq a => a -> a -> Graph a -> Maybe [a]
 tracePathBetween from to g =
     fromPOV from g |>= rootZipper >>= findLoc to |>= trail |>= map value |>= (++ [to])
 
 reparent :: Eq a => Zipper a -> Graph a
 reparent (Zipper g []) = g
-reparent (Zipper g (c:cs)) = Graph (value g) $ (children g) ++ [reparented]
+reparent (Zipper g (c:cs)) = Graph (value g) $ children g ++ [reparented]
     where reparented = reparent (Zipper (crumbToGraph c) cs)
 
 down :: Zipper a -> Maybe (Zipper a)
-down (Zipper (Graph v (k:kids)) crumbs) = Just (Zipper k ((Crumb v [] kids):crumbs))
+down (Zipper (Graph v (k:kids)) crumbs) = Just (Zipper k (Crumb v [] kids:crumbs))
 down _ = Nothing
 
 right :: Zipper a -> Maybe (Zipper a)
-right (Zipper here ((Crumb v lefts (r:rights)):cs)) = Just (Zipper r (shifted:cs))
+right (Zipper here (Crumb v lefts (r:rights):cs)) = Just (Zipper r (shifted:cs))
```

Comment:
I rewrote both lines, adding spaces and substituting parenthesis by `$`.
I still don't like how it looks, but I expect it to be a little more readable.